android/src/app/organicmaps/routing/RoutingBottomMenuController.java

```diff
@@ -294,25 +304,68 @@ private void showRoutingDetails()
   }
 
   @NonNull
-  private static Spanned makeSpannedRoutingDetails(@NonNull Context context, @NonNull RoutingInfo routingInfo)
-
+  private static Spanned makeSpannedRoutingDetails(@NonNull Context context, @NonNull RoutingInfo routingInfo,
+                                                   boolean showTime)
   {
-    CharSequence time = RoutingController.formatRoutingTime(context,
-                                                            routingInfo.totalTimeInSeconds,
-                                                            R.dimen.text_size_routing_number);
+    final SpannableStringBuilder builder = new SpannableStringBuilder();
 
-    SpannableStringBuilder builder = new SpannableStringBuilder();
-    initTimeBuilderSequence(context, time, builder);
-
-    String dot = "\u00A0• ";
-    initDotBuilderSequence(context, dot, builder);
+    if (showTime)
+    {
+      final CharSequence time = RoutingController.formatRoutingTime(context, routingInfo.totalTimeInSeconds,
+                                                                    R.dimen.text_size_routing_number);
+      initTimeBuilderSequence(context, time, builder);
+      final String dot = "\u00A0• ";
+      initDotBuilderSequence(context, dot, builder);
+    }
+    else
+      // If we don't show time then we have place to show "Distance: " label
+      initDistanceBuilderSequence(context, context.getString(R.string.placepage_distance) + ": ", builder);
 
     initDistanceBuilderSequence(context, routingInfo.distToTarget.toString(context), builder);
 
     return builder;
   }
 
-  private static void initTimeBuilderSequence(@NonNull Context context, @NonNull CharSequence time,
+  @NonNull
+  private static Spanned makeSpannedRoutePointsInfo(@NonNull Context context, @NonNull RouteMarkData[] points)
+  {
+    final SpannableStringBuilder builder = new SpannableStringBuilder();
+
+    for(int i=0; i<points.length-1; i++) {
+      RouteMarkData segmentStart = points[i];
+      RouteMarkData segmentEnd = points[i+1];
+      String marker;
+      DistanceAndAzimut dist = Framework.nativeGetDistanceAndAzimuthFromLatLon(segmentStart.mLat, segmentStart.mLon, segmentEnd.mLat, segmentEnd.mLon, 0);
+
+      if (i==0)
+        marker = "🔵";
+      else if (i == 1)
+        marker = "①";
+      else if (i == 2)
+        marker = "②";
+      else if (i == 3)
+        marker = "③";
+      else if (i == 4)
+        marker = "④";
+      else if (i == 5)
+        marker = "⑤";
```

Comment:
https://symbl.cc/en/unicode/blocks/enclosed-alphanumerics/
They are continuous, so can do a trick up to 20:
marker = FromCode((U+2460) + i - 1);

Reply:
Not relevant. Route details UI is inherited from transit info.